functions-and-operators/string-functions.md

```diff
@@ -647,11 +647,71 @@ SELECT INSERT('あああああああ', 2, 3, 'xx');
 
 ### [`INSTR()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_instr)
 
-Return the index of the first occurrence of substring.
+The `INSTR(str, substr)` function is used to get the position of the first occurrence of `substr` in `str`. Each argument can be either a string or a number. This function is the same as the two-argument version of [`LOCATE(substr, str)`](#locate), but with the order of the arguments reversed.
+
+> **Note:**
+>
+> `INSTR(str, substr)` is case-sensitive by default, since [TiDB defaults to using binary collations](/character-set-and-collation.md), which differs from MySQL.
+
+- If either argument is a number, the function treats the number as a string.
+- If `substr` is not in `str`, the function returns `0`. Otherwise, it returns the position of the first occurrence of `substr` in `str`.
+- If either argument is `NULL`, the function returns `NULL`.
+
+Examples:
+
+```sql
+SELECT INSTR("pingcap.com", "tidb");
+
++------------------------------+
+| INSTR("pingcap.com", "tidb") |
++------------------------------+
+|                            0 |
++------------------------------+
+```
+
+```sql
+SELECT INSTR("pingcap.com/tidb", "tidb");
+
++-----------------------------------+
+| INSTR("pingcap.com/tidb", "tidb") |
++-----------------------------------+
+|                                13 |
++-----------------------------------+
+```
+
+```sql
+SELECT INSTR("pingcap.com/tidb", "TiDB");
```

Comment:
Which tidb version do you use? I tried it on my tidb cluster with tidb commit cac449b3370c5, and get different result.

```sql
mysql> SELECT INSTR("pingcap.com/tidb", "TiDB");
+-----------------------------------+
| INSTR("pingcap.com/tidb", "TiDB") |
+-----------------------------------+
|                                13 |
+-----------------------------------+
```

Reply:
```bash
mysql> SELECT TIDB_VERSION()\G
*************************** 1. row ***************************
TIDB_VERSION(): Release Version: v7.1.1-serverless
Edition: Community
Git Commit Hash: 14db49a2968b68ac4cf3e4a59db87ba30a97996d
Git Branch: release-7.1-serverless
UTC Build Time: 2024-01-29 10:05:53
GoVersion: go1.21.6
Race Enabled: false
TiKV Min Version: 6.1.0
Check Table Before Drop: false
Store: tikv
1 row in set (0.24 sec)

mysql> SHOW VARIABLES LIKE 'collation\_%';
+----------------------+-------------+
| Variable_name        | Value       |
+----------------------+-------------+
| collation_connection | utf8mb4_bin |
| collation_database   | utf8mb4_bin |
| collation_server     | utf8mb4_bin |
+----------------------+-------------+
3 rows in set (0.25 sec)

mysql> SELECT INSTR("pingcap.com/tidb", "TiDB");
+-----------------------------------+
| INSTR("pingcap.com/tidb", "TiDB") |
+-----------------------------------+
|                                 0 |
+-----------------------------------+
1 row in set (0.24 sec)
```

Reply:
Currently, the "master" branch of TiDB docs is for TiDB v7.6.0. @jyf111, would you try installing TiDB v7.6 according to [this quick start guide](https://docs.pingcap.com/tidb/dev/quick-start-with-tidb)? Thanks.

Reply:
```bash
mysql> SELECT TIDB_VERSION()\G
*************************** 1. row ***************************
TIDB_VERSION(): Release Version: v7.6.0
Edition: Community
Git Commit Hash: 52794d985ba6325d75a714d4eaa0838d59425eb6
Git Branch: heads/refs/tags/v7.6.0
UTC Build Time: 2024-01-22 14:20:42
GoVersion: go1.21.5
Race Enabled: false
Check Table Before Drop: false
Store: tikv
1 row in set (0.01 sec)

mysql> SHOW VARIABLES LIKE 'collation\_%';
+----------------------+--------------------+
| Variable_name        | Value              |
+----------------------+--------------------+
| collation_connection | utf8mb4_0900_ai_ci |
| collation_database   | utf8mb4_bin        |
| collation_server     | utf8mb4_bin        |
+----------------------+--------------------+
3 rows in set (0.01 sec)

mysql> SELECT INSTR("pingcap.com/tidb", "TiDB");
+-----------------------------------+
| INSTR("pingcap.com/tidb", "TiDB") |
+-----------------------------------+
|                                13 |
+-----------------------------------+
1 row in set (0.00 sec)
```

Um.. But I don't know why `collation_connection` becomes `utf8mb4_0900_ai_ci`.

The latest [doc](https://docs.pingcap.com/tidb/v7.6/system-variables#collation_connection) still shows that the default value of `collation_connection` is `utf8mb4_bin`.

Reply:
Hi @jyf111, thanks for checking the version. 

According to @YangKeao (a TiDB developer), `utf8mb4_0900_ai_ci` was introduced in v7.4 to be compatible with MySQL 8.0 (https://github.com/pingcap/tidb/pull/45650). 

Starting from v7.4.0,  the default collation of `utf8mb4` on TiDB depends on the version of MySQL client, and @YangKeao will create a PR to clarify this issue in the [doc](https://docs.pingcap.com/tidb/v7.6/system-variables#collation_connection).

Reply:
@jyf111 FYI, here is the docPR created by @YangKeao: https://github.com/pingcap/docs/pull/16423/files

Reply:
Great. I have updated the sentences related to case sensitivity and collations. PTAL